src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -43,16 +60,15 @@ public static IMvcCoreBuilder AddMvcCore(this IServiceCollection services)
                 throw new ArgumentNullException(nameof(services));
             }
 
-            ConfigureDefaultServices(services);
-
-            AddMvcCoreServices(services);
-
-            if (setupAction != null)
+            if (setupAction == null)
```

Comment:
Some repositories did this, some other repositories did what you mention. I spoke to @Eilon about it and we set on just throwing in this case. 
Why would you call this with null instead of just calling the other overload? We don't consider this to be an scenario.
If you still want to do this, you only need to pass in ```_=>{}```. instead of null.